mLRS/CommonRx/mavlink_interface_rx.h

```diff
@@ -180,14 +180,17 @@ void MavlinkBase::putc(char c)
         // this will make ArduPilot to response with a NACK:FileNotFound
         // which will make the GCS to quickly jump to normal parameter upload
         if (msg_serial_out.msgid == FASTMAVLINK_MSG_ID_FILE_TRANSFER_PROTOCOL) {
-            uint8_t target_component = msg_serial_out.payload[2];
             uint8_t opcode = msg_serial_out.payload[6];
             char* url = (char*)(msg_serial_out.payload + 15);
-            if ((target_component == MAV_COMP_ID_AUTOPILOT1) && (opcode == MAVFTP_OPCODE_OpenFileRO)) {
-                if (!strcmp(url, "@PARAM/param.pck")) {
+            if (opcode == MAVFTP_OPCODE_OpenFileRO) {
+	        if (!strncmp(url, "@PARAM/param.pck", 16)) {
                     // now fake it to "@xARAM/xaram.xck"
                     url[1] = url[7] = url[13] = 'x';
-                    //return; // for MissionPlanner it doesn't matter what we do, but for QGC?
+		    // Need to recalculate CRC
+		    uint16_t crc = fmav_crc_calculate(&(buf_link_in[1]), FASTMAVLINK_HEADER_V2_LEN - 1);
+		    fmav_crc_accumulate_buf(&crc, msg_serial_out.payload, msg_serial_out.len);
```

Comment:
I need to think more about this, but my first impulse:
- can't line 190&191 be done together by giving the correct length?
- more seious: is line 191 correct? If I'm not mistaken, msg_ser_out.len is the length of the complete frame as it has to be send over the serial so that using it here the calculation actually goes way beyond the frame?

Reply:
Yes, I think 190 and 191 could be combined.  Do you prefer that?  I was following examples in the library.
No, I checked the code in fmav_frame_buf_to_msg().  len is the payload length.  Also, I got the NAK with this code, and not without it, so the calculated checksum must be correct.

Reply:
I was mistaken, haven't done a crc calc function but a function which was reconstructing the extra crc from an unknown message. 
So, I suggest to change line 190&191 to `uint16_t crc = fmav_crc_calculate(&(buf_link_in[1]), FASTMAVLINK_HEADER_V2_LEN - 1 + msg_serial_out.len);`. If you are ok with this If not, leave it as it is.

Reply:
I just realized we can't combine 190 and 191 in this way, because we have modified the payload in msg_serial_out without changing it in buf_link_in.  So I'll leave this part of the code as is.

Reply:
ah, sure, good catch!
maybe you could add a comment // use xxxx because yyy or so :)